Automattic/bugomattic
# Add issue details and task-related state

#### What Does This PR Add/Change?

In short, this PR adds redux state (and corresponding selectors and reducers) for the basic app state required for the core Bugomattic flow (pick feature > pick issue type > see tasks).

To do that, this PR...

- Flattened tasks into their "table" in the normalized reporting config. This helps with tracking tasks in global state.
- Adding an `issueDetails` slice to the redux store -- this is where the info about the issue goes
- Added a `completedTasks` slice to the redux store -- this is how we track the IDs of tasks that have been completed. This lets us provide a "checkbox" experience to the users.
- Adding selectors to the reporting config slice that grab all the relevant tasks at a given point in time (based on the issue details).

I realize there are a lot of files changed, and I'm so sorry! A lot of that is just logistical clean-up. The core changes are themselves not that big. I will leave comments to draw attention to what changes are meaningful!

#### Testing Instructions

Run the unit tests: `yarn test:once`.

I also added some debugging components that let you play around with the fake flow! To start the app locally, just `yarn` to install dependencies, then run `yarn start`.

You can compare what you see to the fake JSON document located at `src/api/local-reporting-config-response.json`.
